src/synthesizer/parametric/morphology.py

```
@@ -375,21 +499,29 @@
+            self.r_eff_mas = r_eff.to("mas").value
         else:
-            raise exceptions.MissingAttribute(
-                "The effective radius must be provided"
+            raise exceptions.IncorrectUnits(
+                "The units of r_eff must have length or angle dimensions"
             )
+        self.r_eff = r_eff
 
+        # Ensure r_eff and x_0, y_0 are in compatible units
+        if not unit_is_compatible(x_0, self.r_eff.units):
+            raise exceptions.InconsistentUnits(
```

Pylint (3.3.7)

[error] Module 'synthesizer.exceptions' has no 'InconsistentUnits' member

(E1101)